src/k8s/pkg/k8sd/features/calico/internal.go

```diff
@@ -0,0 +1,176 @@
+	apiServerEnabled bool
+	autodetectionV4  map[string]any
+	autodetectionV6  map[string]any
+}
+
+func checkEncapsulation(v string) error {
+	switch v {
+	case "VXLAN", "IPIP", "IPIPCrossSubnet", "VXLANCrossSubnet", "None":
+		return nil
+	}
+	return fmt.Errorf("unsupported encapsulation type: %s", v)
+}
+
+func internalConfig(annotations types.Annotations) (config, error) {
+	config := config{
+		encapsulationV4:  defaultEncapsulation,
+		encapsulationV6:  defaultEncapsulation,
+		apiServerEnabled: defaultApiServerEnabled,
+	}
+
+	if v, ok := annotations.Get(annotationAPIServerEnabled); ok {
+		config.apiServerEnabled = v == "true"
+	}
+
+	if v, ok := annotations.Get(annotationEncapsulationV4); ok {
+		if err := checkEncapsulation(v); err != nil {
+			return config, fmt.Errorf("invalid encapsulation-v4 annotation: %w", err)
+		}
+		config.encapsulationV4 = v
+	}
+
+	if v, ok := annotations.Get(annotationEncapsulationV6); ok {
+		if err := checkEncapsulation(v); err != nil {
+			return config, fmt.Errorf("invalid encapsulation-v6 annotation: %w", err)
+		}
+		config.encapsulationV6 = v
+	}
+
+	var autodetectionV4Key string
+	var autodetectionV4Value any
+
+	if v, ok := annotations.Get(annotationAutodetectionV4Firstfound); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Firstfound)
+		}
+		autodetectionV4Key = "firstFound"
+		autodetectionV4Value = v == "true"
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4Kubernetes); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Kubernetes)
+		}
+		autodetectionV4Key = "kubernetes"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4Interface); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Interface)
+		}
+		autodetectionV4Key = "interface"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4SkipInterface); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4SkipInterface)
+		}
+		autodetectionV4Key = "skipInterface"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4CanReach); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4CanReach)
+		}
+		autodetectionV4Key = "canReach"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4Cidrs); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Cidrs)
+		}
+		autodetectionV4Key = "cidrs"
+		autodetectionV4Value = strings.Split(v, ",")
+	}
```

Comment:
let's loop over a `map[string]string{annotationAutodetectionV4CanReach: "cidrs", ...}`?

Reply:
I have also wanted to do this, I'm not sure how to handle `cidrs` and `firstFound` options as they expect values other than string.

Reply:
the value can be a `any`, so it would hold a string, a bool, or a []string just fine.